### PR TITLE
Add dimensionless spin magnitude functions and compute tag

### PR DESCRIPTION
**Pull request by MarloMo**
## Proposed changes

This PR adds the functions and tags used to calculate the dimensionless spin magnitude of a Strahlkorper. Computes the dimensionless spin magnitude from the dimensionful spin magnitude and the Christodoulou mass of a black hole. Adds the DimensionlessSpinMagnitudeCompute tag to EvolveGeneralizedHarmonicWithHorizon.hpp and EvolveGhBinaryBlackHole.hpp.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/ApparentHorizons/StrahlkorperGr.hpp
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: Change to "The argument `strahlkorper` is the surface on which the spin magnitude is computed."

File: src/ApparentHorizons/StrahlkorperGr.hpp
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: `dimensionless spin magnitude \f$\chi\f$ from the`

File: src/ApparentHorizons/StrahlkorperGr.hpp
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: Specifically, this computes `\f$\chi = \frac{S}{M^2}\f$`

**Review by geoffrey4444 — CHANGES_REQUESTED** [review]

File: tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp
Comment (knelli2) [inline tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp]: remove?

File: src/Evolution/Executables/GeneralizedHarmonic/EvolveGeneralizedHarmonicWithHorizon.hpp
Comment (knelli2) [inline src/Evolution/Executables/GeneralizedHarmonic/EvolveGeneralizedHarmonicWithHorizon.hpp:76]: Also add to EvolveGhBinaryBlackHole.hpp? I assume we'll need this quantity in the future.

File: src/ApparentHorizons/Tags.hpp
Comment (knelli2) [inline src/ApparentHorizons/Tags.hpp]: `dimensionless spin magnitude \f$\chi\f$ from the` like in StrahlkorperGr.hpp

**Review by knelli2 — CHANGES_REQUESTED** [review]
The commits should be restructured. I recommend combining
a) the first two commits and
b) everything from the third commit except adding the tags to the executable

into one single commit to keep all related changes/tests together. Then adding the tags to each executable (GHWithHorizon and GhBinaryBlackHole) should get their own commit each.

**Comment by knelli2** [comment]
LGTM. Wait for @geoffrey4444's approval to squash.

Also, when squashing, edit the commit messages so they adhere to [our guidlines](https://spectre-code.org/contributing_to_spectre.html#git-commit-messages).

**Review by geoffrey4444 — DISMISSED** [review]

File: src/ApparentHorizons/StrahlkorperGr.hpp
Comment (nilsdeppe) [inline src/ApparentHorizons/StrahlkorperGr.hpp:483]: While squashing can you add `/// @{` here?

File: src/ApparentHorizons/StrahlkorperGr.hpp
Comment (nilsdeppe) [inline src/ApparentHorizons/StrahlkorperGr.hpp:503]: While squashing can you add `/// @}` here? This makes the dox comment aplly to both functions

**Review by nilsdeppe — COMMENTED** [review]
Quick suggestion. I can merge one squashed and rebase

File: src/ApparentHorizons/StrahlkorperGr.hpp
Comment (MarloMo) [reply src/ApparentHorizons/StrahlkorperGr.hpp:503]: yep , can do, making the changes now

**Comment by knelli2** [comment]
@MarloMo Your second commit message is still too long. It has to be less than 72 characters

**Review by knelli2 — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**
